### ISSUE TITLE
Modals do not trap focus for mobile screen readers

### ISSUE THREAD

**Issue by hrb-hub** [tutao/tutanota#8767]
This was fixed in #7998 but had to be reverted because it broke calendar which used dropdown. Check if safe to reintroduce the fix.

**Estimated: 3h**

## Test notes
- [ ] The topmost modal traps focus
  - [ ] When there's only one modal open
    - [ ] Opening a modal traps focus in modal
    - [ ] Closing modal returns focus to the element used to open it
  - [ ] When there are two or more modals open
    - [ ] Opening a modal traps focus in opened modal
    - [ ] Closing topmost modal returns focus to the element used to open it in its parent modal
- [ ] Overlays behind an open modal should be inert
  - [ ] Open mail editor, write something and close it. This should cause a "draft saved" overlay to appear.
  - [ ] Without dismissing the overlay open the mail editor again (write a new email).
  - [ ] Keep selecting next item until you reach the end.
  - [ ] See that focus is trapped in the mail editor modal.
- [ ] Overlays above an open modal can be interacted with
  - [ ] Enable autoresponder.
  - [ ] Logout and log back in.
  - [ ] See notification overlay with the message "Autoresponder has been activated."
  - [ ] Open the mail editor (on mobile, you'll need to do it quickly before the notification appears).
  - [ ] While the overlay cannot be focused, see #8995, it should not be inert (should be able to trigger its button when not using a screen reader)
